apps/condo/domains/organization/utils/accessSchema.js

```diff
@@ -60,44 +44,84 @@ async function checkRelatedOrganizationPermission (userId, organizationId, permi
+
+    if (uniqOrganizationIds.some(orgId => !isString(orgId))) return false
+
+    for (const id of organizationIds) {
+        const hasAccess = await checkOrganizationPermission(userId, id, permission)
+        if (!hasAccess) return false
     }
-    const organizationFromIds = organizationLinks.map(link => link.from)
 
-    return checkOrganizationsPermission(userId, organizationFromIds, permission)
+    return true
 }
 
-// TODO(nomerdvadcatpyat): use this function where checkRelatedOrganizationPermission and checkOrganizationPermission used together
-async function checkPermissionInUserOrganizationOrRelatedOrganization (userId, organizationId, permission) {
-    if (!userId || !organizationId) return false
-    const hasPermissionInRelatedOrganization = await checkRelatedOrganizationPermission(userId, organizationId, permission)
-    const hasPermissionInUserOrganization = await checkOrganizationPermission(userId, organizationId, permission)
+/**
+ * Check that the user has access in each related organization
+ *
+ * @param userId {string} User.id field
+ * @param organizationIds {Array<string>} array of objects related organizations
+ * @param permission {string} OrganizationEmployeeRole permission key to check for
+ * @return {Promise<boolean>}
+ */
+async function checkRelatedOrganizationsPermission (userId, organizationIds, permission) {
+    if (!userId || !isArray(organizationIds) || isEmpty(organizationIds)) return false
+
+    const uniqOrganizationIds = uniq(organizationIds)
+
+    if (uniqOrganizationIds.some(orgId => !isString(orgId))) return false
 
-    return Boolean(hasPermissionInRelatedOrganization || hasPermissionInUserOrganization)
+    for (const id of organizationIds) {
+        const hasAccess = await checkRelatedOrganizationPermission(userId, id, permission)
+        if (!hasAccess) return false
+    }
+
+    return true
 }
 
 /**
  * Check permission for user to work with multiple objects in case of usage bulk request
+ *
+ * Check that the user has access in each organization (own or related)
+ *
  * @param userId {string} User.id field
  * @param organizationIds {Array<string>} array of objects related organizations
  * @param permission {string} OrganizationEmployeeRole permission key to check for
  * @return {Promise<boolean>}
  */
 async function checkPermissionsInUserOrganizationsOrRelatedOrganizations (userId, organizationIds, permission) {
-    if (!userId || !organizationIds.length) return false
-    const hasPermissionInRelatedOrganizations = await checkRelatedOrganizationsPermission(userId, organizationIds, permission)
-    const hasPermissionInUserOrganizations = await checkOrganizationsPermission(userId, organizationIds, permission)
+    if (!userId || !isArray(organizationIds) || isEmpty(organizationIds)) return false
+
+    const uniqOrganizationIds = uniq(organizationIds)
+
+    if (uniqOrganizationIds.some(orgId => !isString(orgId))) return false
```

Comment:
What the real reason of this check? Correct me if I'm wrong, but are you trying to check if one of the input of organizationIds is not a valid uuid string? If that true, you should add a proper check for uuid pattern via regex or uuid validation utils

Reply:
The idea was to immediately block requests with an empty organizations. But yes, we can immediately check the correctness of the identifier, I’ll add this

Reply:
Done